### PR TITLE
AIP-6692 KFP UI Remove Retry Button

### PR DESCRIPTION
**Pull request by talebzeghmi**
image

### CODE REVIEW

File: frontend/server/handlers/workflow.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return;
+  }
+  if (!namespace) {
+    res.status(400).send('namespace argument is required');
+    return;
+  }
+
+
+  try {
+    await k8sHelper.deleteArgoWorkflow(workflowName, namespace);
```
Comment (alexlatchford) [inline frontend/server/handlers/workflow.ts]: If we just delete a workflow like this and I'd imagine it'd disappear from `etcd`, how does the `metadata-writer` service know to synchronize the state to the KFP metadata store that powers the underlying UI experiences?

I'm guessing that controller is listening to the DELETE event stream from `etcd` and that's how that'd work?

File: frontend/src/lib/Buttons.ts [hunk 2]
```diff
@@ -532,6 +549,24 @@ export default class Buttons {
...
+    ids: string[][],
+    useCurrentResource: boolean,
+    callback: (_: string[], success: boolean) => void,
+  ): void {
+    this._dialogActionHandler(
+      ids,
+      'Do you want to terminate this run? This action cannot be undone. This will terminate any' +
+        ' running pods.',
+      useCurrentResource,
+      id => Apis.runServiceApi.terminateRunDelete(id[0], id[1]).then((x) => {}),
```
Comment (alexlatchford) [inline frontend/src/lib/Buttons.ts]: What does this callback do? 😅 Seems like a no-op. Do we refresh the page or display a success/failure message if the call succeeded or failed?

File: frontend/src/lib/Buttons.ts [hunk 3]
```diff
@@ -113,12 +113,13 @@ export default class Buttons {
     useCurrentResource: boolean,
     callback: (selectedIds: string[], success: boolean) => void,
   ): Buttons {
+    // BUG(talebz): AIP-6692 WFSDK: Disable Retry button
     this._map[ButtonKeys.RETRY] = {
-      action: () => this._retryRun(getSelectedIds(), useCurrentResource, callback),
-      disabled: !useCurrentResource,
-      disabledTitle: useCurrentResource ? undefined : 'Select at least one resource to retry',
+      action: () => window.open('http://analytics.pages.zgtools.net/artificial-intelligence/ai-platform/aip-docs/kubeflow/user_journeys/1_getting_started/index.html'),
```
Comment (alexlatchford) [inline frontend/src/lib/Buttons.ts]: Guessing we need to link to the FAQ article on how to use `Create Run`?

File: frontend/src/lib/Buttons.ts [hunk 4]
```diff
@@ -113,12 +113,13 @@ export default class Buttons {
...
+      disabled: true,
+      disabledTitle: 'AIP Deprecated Retry',
       id: 'retryBtn',
-      title: 'Retry',
+      title: 'AIP Deprecated Retry',
```
Comment (alexlatchford) [inline frontend/src/lib/Buttons.ts]: Do we need to change the title of the button here? Does seem pretty messy to me.

Also have we reported or found any upstream bugs about the underlying root cause?

**Review by alexlatchford — COMMENTED** [review]
Generally looks good to me, definitely better than it was where things were broken and not obvious to users but mostly clarifying behavior from my side :)

File: frontend/server/handlers/workflow.ts (hunk 1 above)
Comment (talebzeghmi) [reply frontend/server/handlers/workflow.ts]: to simplify changes, keeping this one just to removing the RETRY button

File: frontend/src/lib/Buttons.ts (hunk 2 above)
Comment (talebzeghmi) [reply frontend/src/lib/Buttons.ts]: to simplify changes, keeping this one just to removing the RETRY button

File: frontend/src/lib/Buttons.ts (hunk 4 above)
Comment (talebzeghmi) [reply frontend/src/lib/Buttons.ts]: We found it cleaner to remove the RETRY button altogether, and users can use the CLONE button

File: frontend/src/lib/Buttons.ts (hunk 3 above)
Comment (talebzeghmi) [reply frontend/src/lib/Buttons.ts]: removed the Retry button for this PR

File: frontend/src/lib/Buttons.ts [hunk 5]
```diff
@@ -118,7 +118,7 @@ export default class Buttons {
       disabled: !useCurrentResource,
       disabledTitle: useCurrentResource ? undefined : 'Select at least one resource to retry',
       id: 'retryBtn',
-      title: 'Retry',
+      title: 'Retry',
```
Comment (cloudw) [inline frontend/src/lib/Buttons.ts]: Please remove extra spaces at end of this line.

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -55,3 +55,21 @@ build:publish-metadata-writer:
...
+  script:
+    - |
+      docker build \
+        --tag ${IMAGE_REPOSITORY_TAG} \
+        --build-arg COMMIT_HASH=${COMMIT_HASH} \
+        --build-arg DATE="`date -u`" \
+        --file frontend/Dockerfile \
+        .
+    - docker login --username ${DOCKER_USERNAME} --password ${DOCKER_API_KEY} ${DOCKER_REPO_URL}
+    - docker push ${IMAGE_REPOSITORY_TAG}
```
Comment (cloudw) [inline .gitlab-ci.yml]: (minor) maybe add a new line here.

File: frontend/src/pages/RunDetails.tsx [hunk 7]
```diff
@@ -200,16 +200,21 @@ class RunDetails extends Page<RunDetailsInternalProps, RunDetailsState> {
...
+        //       : runIdFromParams
+        //       ? [runIdFromParams]
+        //       : [],
+        //   true,
+        //   () => this.retry(),
+        // )
+        // .newRunFromPipelineVersion(
+        //   () => this.state.runMetadata?.id!,
+        //   () => this.state.runMetadata?.ver!,
+        // )
```
Comment (cloudw) [inline frontend/src/pages/RunDetails.tsx]: Is `.newRunFromPipelineVersion` in comment still needed?

File: frontend/src/pages/RunDetails.tsx [hunk 8]
```diff
@@ -868,9 +873,9 @@ class RunDetails extends Page<RunDetailsInternalProps, RunDetailsState> {
       const actions = buttons.getToolbarActionMap();
       actions[ButtonKeys.TERMINATE_RUN].disabled =
         (runMetadata.status as NodePhase) === NodePhase.TERMINATING || runFinished;
-      actions[ButtonKeys.RETRY].disabled =
-        (runMetadata.status as NodePhase) !== NodePhase.FAILED &&
-        (runMetadata.status as NodePhase) !== NodePhase.ERROR;
+      // actions[ButtonKeys.RETRY].disabled =
```
Comment (cloudw) [inline frontend/src/pages/RunDetails.tsx:873]: Please add the same `// BUG(talebz): AIP-6692 WFSDK: Disable Retry button` comment here for consistency - this is a very long file and context can be easily lost outside this PR.

File: frontend/src/apis/run/api.ts [hunk 9]
```diff
@@ -16,6 +16,8 @@ import * as url from 'url';
 import * as portableFetch from 'portable-fetch';
 import { Configuration } from './configuration';
 
+const request = require('request');
```
Comment (cloudw) [inline frontend/src/apis/run/api.ts]: What is this var for?

**Review by cloudw — CHANGES_REQUESTED** [review]
Thanks for the quick work! Some minor change would make this PR more elegant. Thanks!

File: frontend/src/lib/Buttons.ts (hunk 5 above)
Comment (talebzeghmi) [reply frontend/src/lib/Buttons.ts]: done

File: frontend/src/pages/RunDetails.tsx (hunk 7 above)
Comment (talebzeghmi) [reply frontend/src/pages/RunDetails.tsx]: done

File: .gitlab-ci.yml (hunk 6 above)
Comment (talebzeghmi) [reply .gitlab-ci.yml]: done

File: frontend/src/apis/run/api.ts (hunk 9 above)
Comment (talebzeghmi) [reply frontend/src/apis/run/api.ts]: removed!

File: frontend/src/pages/RunDetails.tsx (hunk 8 above)
Comment (talebzeghmi) [reply frontend/src/pages/RunDetails.tsx:873]: done!

File: .gitlab-ci.yml (hunk 6 above)
Comment (cloudw) [inline .gitlab-ci.yml:75]: Shall we worry about image tag collision with the `build:publish-metadata-writer` step above? Looks like both gitlab ci step runs `docker push ${IMAGE_REPOSITORY_TAG}`

**Review by cloudw — CHANGES_REQUESTED** [review]

File: .gitlab-ci.yml (hunk 6 above)
Comment (talebzeghmi) [reply .gitlab-ci.yml:75]: It's based off of https://github.com/zillow/pipelines/blob/9fbe629a86ca3bef907b81400463418cb0b3f5b4/.gitlab-ci.yml#L63 `ARTIFACT_NAME` see [code link](https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/aip-infrastructure/ci-templates/ci-cd-template/-/blob/main/blocks/docker.yml#L35-40)

also see how the image names are different in the following pipeline:
 https://gitlab.zgtools.net/analytics/artificial-intelligence/ai-platform/sdks/internal/zillow-kfp/-/pipelines/2884129

Comment (cloudw) [reply .gitlab-ci.yml:75]: Understood. Thanks for pointing that out.

**Review by cloudw — APPROVED** [review]

**Pull request merged**